scripts/app.js

```diff
@@ -3,58 +3,77 @@
 ///////////////////////// MUSIC CONSTRUCTOR FUNCTION ///////////////////////
 var piecesArr = [];
 
-function Pieces(musicPiecesObj) {
+function Pieces (musicPiecesObj) {
     this.title = musicPiecesObj.title;
     this.composer = musicPiecesObj.composer;
     this.description = musicPiecesObj.description;
     this.pieceUrl = musicPiecesObj.pieceUrl;
     this.image = musicPiecesObj.image;
     this.recorded = musicPiecesObj.recorded;
-
+    
 }
 
+Pieces.all = [];
+
 ///////////// HANDLEBARS    quick suggestions delay //////////////////////
 
-Pieces.prototype.toHtml = function () {
+Pieces.prototype.toHtml = function() {
     var theTemplateScript = $("#music-template").html();
     var theTemplate = Handlebars.compile(theTemplateScript);
 
-    this.daysAgo = parseInt((new Date() - new Date(this.recorded)) / 60 / 60 / 24 / 1000);
+    this.daysAgo = parseInt((new Date() - new Date(this.recorded))/60/60/24/1000);
     this.performance = `The ${this.composer} was recorded roughly ${this.daysAgo} days ago.`
     return theTemplate(this);
 }
 
-/**
- * Get music pieces from .json file
- */
-$.getJSON('/data/musicpieces.json')
-    .then(function (jsonData) {
-        jsonData.sort(function (a, b) {
-            return (new Date(b.recorded)) - (new Date(a.recorded));
-        });
 
-        jsonData.forEach(function (musicObject) {
-            piecesArr.push(new Pieces(musicObject));
+Pieces.loadAll = function(musicPiecesObjArr) {
+musicPiecesObjArr.sort(function(a,b) {
+  return (new Date(b.recorded)) - (new Date(a.recorded));
+});
+
+musicPiecesObjArr.forEach(function(musicObject) {
+    Pieces.all.push(new Pieces(musicObject));
+});
+};
+
+    // piecesArr.forEach(function(piecesArr) {
+    //     $('#music-articles').append(piecesArr.toHtml());
+    // });
+
+    
+
+ ///////////////////////            JSON CALL           //////////////////////
+ 
+ 
+Pieces.fetchAll = function() {
```

Comment:
you can also pass in a callback function and invoke it on line 59 instead of `piecesView.initMusicPage();`